circuit-types/src/traits.rs

```diff
@@ -669,14 +696,44 @@ impl<const N: usize, T: SecretShareVarType> SecretShareVarType for [T; N] {
 /// but that the verifier does not. The statement is the set of public inputs
 /// and any other circuit meta-parameters that both prover and verifier have
 /// access to
-pub trait SingleProverCircuit {
+pub trait SingleProverCircuit: Sized {
     /// The witness type, given only to the prover, which generates a blinding
     /// commitment that can be given to the verifier
     type Witness: CircuitBaseType;
     /// The statement type, given to both the prover and verifier, parameterizes
     /// the underlying NP statement being proven
     type Statement: CircuitBaseType;
 
+    /// Returns a reference to the proving key for the circuit
+    #[cfg(not(feature = "test-helpers"))]
+    fn proving_key() -> Rc<ProvingKey<SystemCurve>> {
```

Comment:
Should this (and the other references to proving/verifying keys) be `Arc`s? I forget if we have proofs of the same circuit being generated on multiple threads